### PR TITLE
Feat/xd-109: Backend Error Handling 

### PR DESCRIPTION
**Pull request by Sabo2k**
Based on the [Prisma Documentation](https://www.prisma.io/blog/nestjs-prisma-error-handling-7D056s1kOop2), I've added an Exception Filter to the Case Service. 

It should provide a more readable and user friendly error message for each HTTP request. 

You can toggle the messages by setting ```DEBUG_MODE``` in your ```.env``` file to ```true```

### CODE REVIEW

File: libs/cases/backend/case-management/src/lib/services/case.service.ts [hunk 1]
```diff
@@ -23,12 +30,15 @@ export class XdCaseService {
 	 */
 	public getAllCases(): Observable<ICaseResponse[]> {
 		return from(this.prismaService.case.findMany()).pipe(
-			map((items) =>
-				items.map((item) => ({
+			map((items) => {
+				if (items.length === 0) {
+					throw new NotFoundException('No Cases found.');
```
Comment (KonsumGandalf) [inline libs/cases/backend/case-management/src/lib/services/case.service.ts]: Why should this be an exception? When no case has been created yet, there should Error be thrown

File: libs/cases/backend/case-management/src/lib/services/case.service.ts [hunk 2]
```diff
@@ -23,12 +30,15 @@ export class XdCaseService {
...
+				}
+				return items.map((item) => ({
 					...item,
 					overdue: Date.now() > new Date(item.dueDate).getTime(),
```
Comment (KonsumGandalf) [inline libs/cases/backend/case-management/src/lib/services/case.service.ts]: For sth like this `defaults` of `lodash` might be better, but it is **not** necessary to change it here, resolve this comment own your own.

File: libs/cases/backend/case-management/src/lib/services/case.service.ts [hunk 3]
```diff
@@ -90,6 +105,14 @@ export class XdCaseService {
 				...item,
 				overdue: Date.now() > new Date(item.dueDate).getTime(),
 			})),
+			catchError((error) => {
+				if (error instanceof Prisma.PrismaClientKnownRequestError) {
+					if (error.code === 'P2025') {
+						throw new NotFoundException(`Case with ID ${id} does not exist.`);
+					}
+				}
+				throw new ConflictException('An error occurred while deleting the case.');
```
Comment (KonsumGandalf) [inline libs/cases/backend/case-management/src/lib/services/case.service.ts]: Pls be consitent with the Errors which are used. Why don't you use BadRequest Exception as well?

File: libs/cases/backend/case-management/src/lib/services/case.service.ts [hunk 4]
```diff
@@ -71,6 +81,11 @@ export class XdCaseService {
 				...item,
 				overdue: Date.now() > new Date(item.dueDate).getTime(),
 			})),
+			catchError((error) => {
+				throw new BadRequestException(
+					'Failed to create Case. Please check the provided data.',
```
Comment (KonsumGandalf) [inline libs/cases/backend/case-management/src/lib/services/case.service.ts]:
```suggestion
					'Failed to create a case.',
```

The input is validated so it should be correct already

If you want to add 'real handling' you can directly print the console.log(caseData) in the next sentence

File: package.json [hunk 5]
```diff
@@ -36,6 +36,7 @@
 		"class-validator": "^0.14.1",
 		"dotenv": "^16.4.5",
 		"echarts": "^5.5.0",
+		"express": "^4.19.2",
```
Comment (KonsumGandalf) [inline package.json]: I dont think you need to add this dependency. Pls try to do the statement `const response = context.getResponse<Response>();` without the type it should be automatically interfered. 
Add a large package just for a type seems to be an overhead to me.

If it is not possible *ignore this comment* and resolve it.

File: package.json [hunk 6]
```diff
@@ -80,6 +81,7 @@
 		"@swc-node/register": "~1.9.0",
 		"@swc/core": "~1.5.0",
 		"@swc/helpers": "~0.5.11",
+		"@types/express": "^4.17.21",
```
Comment (KonsumGandalf) [inline package.json]: U dont even use the package.

**Review by KonsumGandalf — CHANGES_REQUESTED** [review]

**Comment by Sabo2k** [comment]
For example, trying to delete a non-existing case gives a clearer error message instead of internal server error: error-messagepng

**Pull request closed without merging**